The hostname may not necessarily be a valid SNI name (e.g. builder-edge-x86).
If it is not a valid SNI name, the tests fail. Just hardcode some host string
instead to workaround that.

This fixes the tests on the builders.

diff -upr guile-gnutls-3.7.14.orig/guile/tests/anonymous-auth.scm guile-gnutls-3.7.14/guile/tests/anonymous-auth.scm
--- guile-gnutls-3.7.14.orig/guile/tests/anonymous-auth.scm	2023-08-05 22:25:07.589333105 +0200
+++ guile-gnutls-3.7.14/guile/tests/anonymous-auth.scm	2023-08-05 22:25:33.116064123 +0200
@@ -79,7 +79,7 @@
        (let ((client (make-session connection-end/client)))
          (set-session-priorities! client priorities)
          (set-session-server-name! client
-                                   server-name-type/dns (gethostname))
+                                   server-name-type/dns "someserver")
          (set-session-transport-fd! client (port->fdes (car socket-pair)))
          (set-session-credentials! client (make-anonymous-client-credentials))
          (set-session-dh-prime-bits! client 1024)
diff -upr guile-gnutls-3.7.14.orig/guile/tests/premature-termination.scm guile-gnutls-3.7.14/guile/tests/premature-termination.scm
--- guile-gnutls-3.7.14.orig/guile/tests/premature-termination.scm	2023-08-05 22:25:07.589333105 +0200
+++ guile-gnutls-3.7.14/guile/tests/premature-termination.scm	2023-08-05 22:25:22.522704084 +0200
@@ -77,7 +77,7 @@
          (close-port (cdr socket-pair))           ;close the server end
          (set-session-priorities! client priorities)
          (set-session-server-name! client
-                                   server-name-type/dns (gethostname))
+                                   server-name-type/dns "servername")
          (set-session-transport-fd! client (port->fdes (car socket-pair)))
          (set-session-credentials! client (make-anonymous-client-credentials))
          (set-session-dh-prime-bits! client 1024)
